build-tools/packages/build-tools/src/fluidBuild/tasks/workers/tscWorker.ts

```diff
@@ -2,85 +2,21 @@
-		if (configFileName) {
-			commandLine = ts.getParsedCommandLineOfConfigFile(configFileName, commandLine.options, {
-				...ts.sys,
-				getCurrentDirectory: () => cwd,
-				onUnRecoverableConfigFileDiagnostic: (diagnostic: tsTypes.Diagnostic) => {
-					diagnostics.push(diagnostic);
-				},
-			});
-		} else {
-			throw new Error("Unknown config file in command line");
-		}
-	}
-
-	let code = ts.ExitStatus.DiagnosticsPresent_OutputsSkipped;
-	if (commandLine && diagnostics.length === 0) {
-		const incremental = !!(commandLine.options.incremental || commandLine.options.composite);
-		const param = {
-			rootNames: commandLine.fileNames,
-			options: tscUtils.convertOptionPaths(commandLine.options, cwd, path.resolve),
-			projectReferences: commandLine.projectReferences,
-		};
-		const program = incremental ? ts.createIncrementalProgram(param) : ts.createProgram(param);
-
-		diagnostics.push(...program.getConfigFileParsingDiagnostics());
-		diagnostics.push(...program.getSyntacticDiagnostics());
-		diagnostics.push(...program.getOptionsDiagnostics());
-		diagnostics.push(...program.getGlobalDiagnostics());
-		diagnostics.push(...program.getSemanticDiagnostics());
-
-		const emitResult = program.emit();
-		diagnostics.push(...emitResult.diagnostics);
+import { tsCompile } from "../../../common/tsCompile.js";
+import { fluidTscRegEx } from "../../../common/tscUtils.js";
+import type { WorkerExecResult, WorkerMessage } from "./worker.js";
 
-		if (emitResult.emitSkipped && diagnostics.length > 0) {
-			code = ts.ExitStatus.DiagnosticsPresent_OutputsSkipped;
-		} else {
-			code =
-				diagnostics.length !== 0
-					? ts.ExitStatus.DiagnosticsPresent_OutputsGenerated
-					: ts.ExitStatus.Success;
-		}
-	}
-
-	if (diagnostics.length > 0) {
-		const sortedDiagnostics = ts.sortAndDeduplicateDiagnostics(diagnostics);
-
-		const formatDiagnosticsHost: tsTypes.FormatDiagnosticsHost = {
-			getCurrentDirectory: () => ts.sys.getCurrentDirectory(),
-			getCanonicalFileName: tscUtils.getCanonicalFileName,
-			getNewLine: () => ts.sys.newLine,
-		};
+export async function compile(msg: WorkerMessage): Promise<WorkerExecResult> {
+	return { code: tsCompile(msg) };
+}
 
-		// TODO: tsc has more complicated summary than this
-		if (commandLine?.options?.pretty !== false) {
-			console.log(
-				ts.formatDiagnosticsWithColorAndContext(sortedDiagnostics, formatDiagnosticsHost),
-			);
-			console.log(
-				`${ts.sys.newLine}Found ${sortedDiagnostics.length} error${
-					sortedDiagnostics.length > 1 ? "s" : ""
-				}.`,
-			);
-		} else {
-			console.log(ts.formatDiagnostics(sortedDiagnostics, formatDiagnosticsHost));
-		}
+export async function fluidCompile(msg: WorkerMessage): Promise<WorkerExecResult> {
+	const commandMatch = fluidTscRegEx.exec(msg.command);
+	if (!commandMatch) {
+		throw new Error(`worker command not recognized: ${msg.command}`);
 	}
-	return { code };
+	const command = `tsc${msg.command.slice(fluidTscRegEx.lastIndex)}`;
```

Comment:
We confirmed this works, right? From [docs](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/lastIndex) on `lastIndex: 

> This property is set only if the regular expression instance used the g flag to indicate a global search, or the y flag to indicate a sticky search.

I don't see either flag in the regex, but I'm also not sure I'm parsing the docs correctly (is the "when exec() is called..." _in addition_ to the flag condition, or alternative?).

Reply:
We hadn't - I thought it would and was going to confirm this morning. :) I'm just going to simplify to `string.match` and `string.replace` and fully confirm. Good callout.